### PR TITLE
Feature C3D

### PR DESCRIPTION
**Pull request by DavidMansolino**
This PR adds the possibility to visualize C3D files directly in Webots.

Current state: https://youtu.be/bJtCoR8DvOA

### CODE REVIEW

**Comment by DavidMansolino** [comment]
I finally found that there is no way to know if a marker is a virtual marker or not:
> By default, there is no way to distinguish a real marker from a virtual marker. However a list of markers introduced by Vicon is known to contains data to represent reference frames (as quatro of markers) or center of mass (also projected on the ground). When found these markers are set as virtual markers.

But I found the list of known virtual markers:
https://github.com/Biomechanical-ToolKit/BTKCore/blob/d4c03aa9e354be16265d0efe0815c09b35abc642/Code/BasicFilters/btkSeparateKnownVirtualMarkersFilter.cpp#L54

**Comment by DavidMansolino** [comment]
test me

**Comment by fabienrohrer** [comment]
test me

**Comment by DavidMansolino** [comment]
I am now able to get the 3D ground reaction forces:
c3d_viewer

But I am still having issues getting their current origins.

**Comment by DavidMansolino** [comment]
I found a formula to compute the positions, I will give it a try: http://c-motion.com/v3dwiki/index.php?title=Tutorial:_Force_Platforms

**Comment by DavidMansolino** [comment]
The GRF is now displayed both in the graphs and 3D view:
[VIDEO](http://www.youtube.com/watch?v=JNV3bamw1iU)

**Comment by fabienrohrer** [comment]
this looks really good!

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
test me

**Comment by DavidMansolino** [comment]
Documentation of Vicon angle data:
  - https://www.vicon.com/faqs/software/what-are-the-upper-body-segment-angles-from-plug-in-gait
  - https://www.vicon.com/faqs/software/what-are-the-lower-body-segment-angles-from-plug-in-gait

**Comment by DavidMansolino** [comment]
For the reviewer:
  - As said in the readme file do not review the ``c3d.py`` file.
  - I did add a few simple example C3D files, but if you want a more representative and complete one please contact me on discord: https://discordapp.com/invite/nTWbN9m

👍

File: projects/humans/c3d/protos/C3dGrf.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#VRML_SIM R2019b utf8
+# tags: static, hidden
+
+PROTO C3dGrf [
```
Comment (omichel) [inline projects/humans/c3d/protos/C3dGrf.proto]: I would rename this proto `C3dGroundReactionForce` which is more understandable for the novice.

**Comment by omichel** [comment]
When editing fields in the `C3dViewer` node, such as `bodyHeight`, I am getting this warning in the Webots console (Windows 10):
```
[javascript] ReferenceError: Can't find variable: QWebChannel (webots.js:61)
```

**Comment by omichel** [comment]
Also the MSYS console display a lot of warnings:
```
Warning: QRegularExpressionPrivate::doMatch(): called on an invalid QRegularExpression object
```

**Comment by DavidMansolino** [comment]
I can reproduce both issues:
> Warning: QRegularExpressionPrivate::doMatch(): called on an invalid QRegularExpression object

This warnings is displayed in MSYS2 after running and saving the world, when loading the world.

> [javascript] ReferenceError: Can't find variable: QWebChannel (webots.js:61)

I can reproduce this error with the ``e-puck.wbt`` world too when changing the ``version`` field (or any regenerative field). This seems to be a problem with our html robot-windows when the robot itself is regenerated.

**Comment by DavidMansolino** [comment]
Was wrongly merged with develop

**Pull request closed without merging**